lib/Doctrine/CouchDB/CouchDBClient.php

```diff
@@ -145,13 +146,38 @@ public function getUuids($count = 1)
     /**
      * Find a document by ID and return the HTTP response.
      *
-     * @param  string $id
+     * If $allOpenRevs is true, documents for all leaf revisions are
+     * retrieved and $revisions parameter is ignored. If $allOpenRevs
+     * is false and $revisions is not null, only the specified leaf revisions
+     * are retrieved.
+     *
+     * @param string $id
+     * @param bool $allOpenRevs
+     * @param array $revisions
      * @return HTTP\Response
      */
-    public function findDocument($id)
-    {
-        $documentPath = '/' . $this->databaseName . '/' . urlencode($id);
-        return $this->httpClient->request( 'GET', $documentPath );
+    public function findDocument(
```

Comment:
Yes the response changes slightly.

1. **Without `allOpenRevs` or `revisions`**
To access the doc: `$response->body`.

2. **With `allOpenRevs` or `revisions`**
 1. To access the docs for non-missing revisions: `$response->body[index]['ok']`.
 2. For missing revisions: `$response->body[index]` is `array('missing' => 'rev')`.

To get a better idea see the `testFindDocument()` method. Let me know if i should add a new method to support the fetching of documents for multiple revisions. If yes, what's desired, a single method supporting these two options or separate ones to fetch all and specified revisions separately.